src/inet/TCPEndPoint.h

```diff
@@ -522,7 +522,7 @@ class DLL_EXPORT TCPEndPoint : public EndPointBasis<TCPEndPoint>
     /**
      * Size of the largest TCP packet that can be received.
      */
-    constexpr static size_t kMaxReceiveMessageSize = System::PacketBuffer::kMaxSizeWithoutReserve;
+    constexpr static size_t kMaxReceiveMessageSize = System::PacketBuffer::kMaxAllocSize;
```

Comment:
Could you also explain this change? I believe that the "withoutreserve" vs maxsize is a bit messy, however overall this seems to change a constant with something else, so it is a change in existing behaviourt even for UDP. I would like to make sure it is correct.